db/migrate/20171026082635_add_task_action.rb

```diff
@@ -0,0 +1,14 @@
+class AddTaskAction < ActiveRecord::Migration
+  def up
+    add_column :foreman_tasks_tasks, :action, :string
+
+    ::ForemanTasks::Task.where(:action => nil).find_each do |task|
+      task.action = task.format_input
```

Comment:
This can be pretty expensive operation, that would slow down the whole migration a lot. I wonder, if we should do it here? Perhaps a rake task to calculate this, if anyone wanted that, but in general, I'm not sure many people would want that for older tasks: if the did, they could run the rake task.

Reply:
The way the UI is now, without actually computing the changes all the old tasks would have blank action column in the tasks index. We could fallback to generating this stuff on demand when the user displays the task, but it might break the search.

Reply:
:+1: for fallback